nixos/modules/services/monitoring/uptime-kuma.nix

```diff
@@ -0,0 +1,111 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.uptime-kuma;
+  settingsToEnv = mapAttrsToList (name: value: "${name}=${value}");
+in {
+
+  options = {
+    services.uptime-kuma = {
+      enable = mkEnableOption "Uptime Kuma";
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/uptimekuma/";
+        description = "The directory where Uptime Kuma stores its data files.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "uptimekuma";
+        description = "User account under which Uptime Kuma runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "uptimekuma";
+        description = "Group under which Uptime Kuma runs.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        example = literalExpression "pkgs.uptime-kuma";
+        defaultText = "pkgs.uptime-kuma";
+        description = ''
+          Uptime Kuma package to use.
+        '';
+      };
+
+      settings = lib.mkOption {
+        type =
+          lib.types.submodule { freeformType = with lib.types; attrsOf str; };
+        default = { };
+        description = ''
+          Additionnal configuration for Uptime Kuma, see
+          <link xlink:href="https://github.com/louislam/uptime-kuma/wiki/Environment-Variables"/>
```

Comment:
`Additional` typo ; Prefer `lib.mdDoc` for documentation (using Markdown).

Reply:
Thanks